src/components/PauseSubscriptionPopup/PauseSubscriptionPopup.js

```diff
@@ -0,0 +1,282 @@
+              switchDirection: toOffer.switchDirection,
+              algorithm: toOffer.algorithm
+            }
+          })
+        );
+        setIsLoading(false);
+        setStep(STEPS.CONFIRMATION);
+      } else {
+        window.dispatchEvent(
+          new CustomEvent('MSSDK:switch-popup-action-failed', {
+            detail: { reason: resp.errors[0] }
+          })
+        );
+        setError(true);
+        setIsLoading(false);
+      }
+    } catch {
+      window.dispatchEvent(new CustomEvent('MSSDK:switch-popup-action-failed'));
+      setError(true);
+      setIsLoading(false);
+    }
+  };
+
+  const closePopupAndRefresh = () => {
+    hideInnerPopup();
+    updateList();
+  };
+
+  if (isPopupLoading) {
+    return (
+      <InnerPopupWrapper
+        steps={2}
+        popupTitle={t('Pausing plan')}
+        currentStep={1}
+      >
+        <SkeletonWrapper
+          showChildren={false}
+          height={200}
+          width={450}
+          margin="auto"
+        />
+      </InnerPopupWrapper>
+    );
+  }
+
+  return (
+    <InnerPopupWrapper
+      steps={isPartOfCancellationFlow ? 3 : 2}
+      popupTitle={t('Pausing plan')}
+      currentStep={
+        isPartOfCancellationFlow ? STEPS_NUMBERS[step] + 1 : STEPS_NUMBERS[step]
+      }
+    >
+      {step === STEPS.PAUSE_DETAILS && !isError && (
+        <>
+          <ContentStyled>
+            <ImageWrapper>
+              <SubscriptionIconStyled
+                period={fromOffer.period}
+                showLabel="Current"
+                gray
+              />
+              <ArrowStyled />
+              <SubscriptionIconStyled isPaused showLabel="Paused" />
+            </ImageWrapper>
+            <TitleStyled step={step} textTransform="capitalize">
+              {t('Subscription pause')}
+            </TitleStyled>
+            <TextStyled>
+              <Trans i18nKey="pausesubscriptionpopup-info">
+                Your subscription will be paused for{' '}
+                {{
+                  pausePeriod: periodMapper[toOffer.period].chargedForEveryText
+                }}{' '}
+                starting{' '}
+                <strong>
+                  {{ pauseStartingDate: dateFormat(fromOffer.expiresAt) }}
+                </strong>
+                . During the subscription pause period, you will not be charged.
+                Your subscription will be automatically resumed after a{' '}
+                {{
+                  pausePeriod: periodMapper[toOffer.period].chargedForEveryText
+                }}{' '}
+                and you will continue to be charged
+                <strong>
+                  {' '}
+                  {{
+                    currencySymbol:
+                      currencyFormat[fromOffer.nextPaymentCurrency]
+                  }}
+                  {{
+                    currentPrice: formatNumber(fromOffer.nextPaymentPrice)
+                  }}
+                </strong>{' '}
```

Comment:
this part is messy and quite hard to ready. Could we move all variables (like `currencyFormat[fromOffer.nextPaymentCurrency]`) to separate variables?